src/asciidoc/docs/guidelines/frontendUploading.adoc

```diff
@@ -0,0 +1,108 @@
+= Uploading files from the frontend
+:page-slug: frontend-uploading
+:page-description: How to enable users to upload files to the CDN from a Koji's frontend.
+
+Some templates provide users with the ability to upload files from a Koji’s frontend.
+For example, a public image board.
+When developing such a template, the recommended way to upload files is to use the `generateSignedUploadRequest` method from the <<withkoji-database-package#.generateSignedUploadRequest, @withkoji/database-sdk>> package.
+
+This method enables the file to be uploaded directly from the Koji’s frontend to the Koji CDN, for optimal performance.
+Other methods, such as `uploadFile`, would require the file to be uploaded to the project’s backend server and then to the CDN.
+
+To upload files with `generateSignedUploadRequest`, you configure the backend to acquire a presigned URL, which grants temporary access for uploading a file.
+Then, you configure the frontend to retrieve a presigned URL from the backend and use it to upload a file.
+
+== Prerequisites
+
+This guide covers advanced topics.
+It is suggested that you are familiar with the following topics before proceeding.
+
+* React and ES6 basics.
+* The Koji editor, <<instant-remixing#, remix process>>, and the <<withkoji-vcc-package,@withkoji/vcc>> package.
+* https://expressjs.com/[Express.js] web framework.
+* Backend communication and the <<withkoji-database-package,@withkoji/database-sdk>> package.
+* Performing HTTP requests (this example uses https://github.com/axios/axios[axios]).
+
+== Configuring the backend
+
+To upload from the frontend, you first need to acquire a presigned URL by using the Koji database SDK.
+
+In a backend service for your Koji project, set up an Express.js server with a POST method to the `/get-upload-url` endpoint.
+From this endpoint, call the following function.
+
+[source, javascript]
+----
+import Database from '@withkoji/database';
+
+const uploadImage = async (req, res) => {
+  try {
+    const { filename } = req.body;
+
+    const db = new Database({
+      projectId: res.locals.KOJI_PROJECT_ID,
+      projectToken: res.locals.KOJI_PROJECT_TOKEN,
+    });
+
+    const data = await db.generateSignedUploadRequest(filename);
+
+    res.status(200).json({ data });
+  } catch (err) {
+    console.log('err', err);
+    res.status(500).json({ err: err.toString() });
+  }
+};
+
+export default uploadImage;
+----
+
+== Configuring the frontend
+
+From your frontend service, call the `/get-upload-url` endpoint to retrieve the presigned URL.
+
+[source,javascript]
+----
+const { data: { data: { signedRequest, url } } = {} } = await axios({
```

Comment:
Might be good to show the structure of the data that is returned along with a brief explanation. While it's okay that this thing feels like magic, it would be helpful for developers to know that this is really just a wrapper around an S3 method and if they see that it's similar to what they've used before may grok it more quickly.